src/sinks/mqtt/config.rs

```diff
@@ -0,0 +1,120 @@
+use codecs::JsonSerializerConfig;
+use rumqttc::{MqttOptions, Transport};
+use serde::{Deserialize, Serialize};
+use snafu::ResultExt;
+
+use crate::{
+    codecs::EncodingConfig,
+    config::{AcknowledgementsConfig, GenerateConfig, Input, SinkConfig, SinkContext},
+    sinks::{
+        mqtt::sink::{TlsSnafu, MqttConnector, MqttError, MqttSink},
+        Healthcheck, VectorSink,
+    },
+    tls::{MaybeTlsSettings, TlsEnableableConfig},
+};
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct MqttSinkConfig {
+    pub host: String,
+    #[serde(default = "default_port")]
+    pub port: u16,
+    pub user: String,
+    pub password: String,
+    #[serde(default = "default_client_id")]
+    pub client_id: String,
+    #[serde(default = "default_keep_alive")]
+    pub keep_alive: u16,
+    #[serde(default = "default_clean_session")]
+    pub clean_session: bool,
+    pub tls: Option<TlsEnableableConfig>,
+    pub topic: String,
+    pub encoding: EncodingConfig,
+}
+
+const fn default_port() -> u16 {
+    1883
+}
+
+fn default_client_id() -> String {
+    "vector".into()
+}
+
+const fn default_keep_alive() -> u16 {
+    60
+}
+
+const fn default_clean_session() -> bool {
+    false
+}
+
+impl GenerateConfig for MqttSinkConfig {
+    fn generate_config() -> toml::Value {
+        toml::Value::try_from(Self {
+            host: "localhost".into(),
+            port: default_port(),
+            user: "admin".into(),
+            password: "secret".into(),
```

Comment:
Wondering if it would make sense to leave these as empty strings since authentication is optional? 🤔

Reply:
Fixed